libzfs.pyx

```diff
@@ -2255,6 +2256,19 @@ cdef class ZFSVdev(object):
         def __get__(self):
             return self.nvlist.get(zfs.ZPOOL_CONFIG_GUID)
 
+    property name:
+        def __get__(self):
+            cdef char *mntpt
+            mntpt = libzfs.zpool_vdev_name(
+                <libzfs.libzfs_handle_t *>self.root,
+                <libzfs.zpool_handle_t *>self.zpool.handle,
+                self.nvlist.handle,
+                libzfs.VDEV_NAME_TYPE_ID)
```

Comment:
We should add checks for whether the libzfs handle and zpool handles are NULL or closed and possibly raise an ZFSException if they are.

Reply:
For example, if for some reason libzfs_fini() is called on a handle then all ZFS dataset handles allocated using it will be freed.

Reply:
@anodos325 sure however we have `zpool` and `root` usages throughout the class and it seems other classes follow the same pattern where the initializer is responsible for making sure we have valid handles..
Should I still go ahead and an explicit check here? I was thinking of perhaps adding a check in the constructor but we are using `__new__` instead in some places where the check would be missed..

Reply:
Right, this is a design flaw since it causes a NULL dereference and crash. 
@yocalebo is working on a fix for this and so maybe some coordination needs to happen regarding what to do here.